awx/ui/src/components/Schedule/shared/ScheduleForm.js

```diff
@@ -562,14 +563,25 @@ function ScheduleForm({
               errors.endDate = t`Please select an end date/time that comes after the start date/time.`;
             }
 
+            if (
+              end === 'onDate' &&
+              DateTime.fromISO(endDate)
+                .diff(DateTime.fromISO(startDate), 'days')
+                .toObject().days < 7
+            ) {
+              const rule = new RRule(buildRuleObj(values));
+              if (rule.all().length === 0) {
+                errors.startDate = t`Please select a valid start and end date/time.`;
```

Comment:
Can we make the error message a little more informative? Maybe something like "Selected frequency does not occur between start and end dates"

Reply:
It would also be nice to add this one as a form-level error, rather than field-level on the startDate... though I don't remember the specifics of how we normally do that... ?